### ISSUE TITLE
[Bug] quantize composite path truncates instead of rounding: ~50% of elements differ from torch and from ttnn's own fused path

### ISSUE THREAD

**Issue by kinginu** [tenstorrent/tt-metal#51304]
### Describe the bug

`ttnn.quantize` returns different results depending on whether the zero-point is passed as a Python `int` or as a 1-element tensor. The tensor form disagrees with `torch.quantize_per_tensor` on roughly **50% of elements**, each off by exactly 1 LSB.

The composite fallback truncates toward zero where torch — and ttnn's own fused QUANT LLK path — round to nearest even.

Nothing aborts, so the existing PCC-based tests pass. The result is silently wrong.

### To Reproduce

```python
import torch, ttnn

device = ttnn.open_device(device_id=0)
torch.manual_seed(0)

x = torch.rand(1, 1, 1024, 1024, dtype=torch.float32) * 4 - 2
scale, zero_point = 0.02, 7

tt_x = ttnn.from_torch(x, dtype=ttnn.float32, layout=ttnn.TILE_LAYOUT, device=device)
zp_t = ttnn.from_torch(torch.tensor([zero_point], dtype=torch.int32),
                       dtype=ttnn.int32, layout=ttnn.TILE_LAYOUT, device=device)

golden = torch.quantize_per_tensor(x, scale, zero_point, torch.qint8).int_repr().to(torch.int32)

for name, zp in (("scalar zero-point", zero_point), ("tensor zero-point", zp_t)):
    got = ttnn.to_torch(ttnn.quantize(tt_x, scale, zp)).to(torch.int32)
    n = (got != golden).sum().item()
    print(f"{name:20s} differ={n} ({100.0 * n / golden.numel():.4f}%)")
```

Measured on Blackhole P300:

```
scalar zero-point    differ=3       (0.0003%)
tensor zero-point    differ=523576  (49.9321%)
```

Every difference is exactly ±1.

The per-channel composite is affected the same way:

```
per-channel, tensor zero-point   differ=32890 (50.1862%)
per-channel, scalar zero-point   differ=0     (0.0000%)
```

### Expected behavior

The op should not depend on how the zero-point is spelled. `torch.quantize_per_tensor` rounds to nearest even, and so does ttnn's own fused path, so the composite fallback should too.

This matters because the quantization ops were added specifically to "reach feature parity with PyTorch's quantization API" (#20290), and the tensor form of the zero-point is what a torch-exported quantized model produces.

### Root cause

Both paths compute the same expression but finish differently.

**Fused path** — `ckernel_sfpu_quant.h`, `quant_init`:

```cpp
TTI_SFPMAD(LREG0, LREG1, LREG2, LREG0, 0);          // x * (1/s) + zp
TTI_SFP_STOCH_RND(sfpi::SFPSTOCHRND_RND_EVEN, ...); // round to nearest even, saturate to int8
```

**Composite path** — `quantization.cpp`:

```cpp
return ttnn::typecast(
    ttnn::add(input_scaled, zero_point_full, ...),
    c_dtype);                                        // <-- truncates
```

The fp32 -> int32 typecast has no rounding term on Wormhole/Blackhole. From `tt_metal/hw/ckernels/blackhole/metal/llk_api/llk_sfpu/ckernel_sfpu_typecast.h`, `calculate_typecast_fp32_to_int32`:

```
SFPEXEXP -> SFPEXMAN -> SFPSHFT(exp - 23) -> two's complement if negative
```

It extracts the mantissa and shifts, i.e. a C-style truncation toward zero. There is no hardware fp32 -> int32 round-to-nearest instruction on these architectures: `SFPCAST_MOD1_FP32_TO_SM32_RNE` exists only under `#if __riscv_xtttensixqsr` in `runtime/sfpi/include/sfpi_constants.h`, so the same `ttnn.typecast` rounds on Quasar and truncates on Wormhole/Blackhole.

For a uniformly distributed fractional part, truncation and round-to-nearest disagree half the time, which matches the measured 49.93%.

### Affected call sites

All four composite paths that end in a float -> integer narrowing:

| file:line (on `6b291d0c107`) | path |
|---|---|
| `quantization.cpp:233` | `quantize`, per-channel, tensor zero-point |
| `quantization.cpp:281` | `quantize`, per-tensor, float scale + tensor zero-point |
| `quantization.cpp:305` | `quantize`, per-tensor, tensor scale + tensor zero-point |
| `quantization.cpp:420` | `requantize`, per-channel, all-tensor |

### Please complete the following environment information

- OS: Ubuntu 22.04
- Version of software: reproduced on `6b291d0c107`; the relevant code is unchanged as of `66d22c1f3a9` (current `main` at the time of filing)
- Hardware: Blackhole P300

### Additional context

The fix does not cost a pass. `binary_ng` runs post-activations in DST before the pack (`PROCESS_POST_ACTIVATIONS` in `eltwise_binary*.cpp`), and it already folds the output narrowing into the same pass when `binary::utils::is_typecast(a_dtype, c_dtype)` holds (`binary_ng_program_factory.cpp`). So passing `ROUND` as a post-activation and letting the `add` produce the integer dtype directly **removes** the separate typecast op.

Measured on Blackhole, 2048x2048:

| call | current | with the fix | |
|---|---|---|---|
| `quantize(x, 0.02, zp_tensor)` | 0.327 ms | 0.237 ms | 1.38x |
| `quantize(x, s_vec, zp_vec, axis=-1)` | 0.971 ms | 0.872 ms | 1.11x |

**Faster and correct**, because the separate `typecast` op goes away.

Note this issue is only about truncation vs round-to-nearest-even. The composite path also does not saturate to the int8 range the way the QUANT LLK does; that is a separate behavioural question and is deliberately out of scope here.

I have a patch and tests ready and will open a PR.
